service/history/workflow/update/registry.go

```diff
@@ -211,7 +216,9 @@ func (r *registry) UpdateFromStore() {
 			)
 		} else if updInfo.GetCompletion() != nil {
 			if upd := r.updates[updID]; upd != nil {
-				_ = upd.advanceTo(stateCompleted)
+				if upd.state != stateProvisionallyCompleted {
+					_ = upd.advanceTo(stateCompleted)
+				}
```

Comment:
It looks like this change and the one above are intended to fix bugs -- can you add a test demonstrating the bugs, or outline what the test would look like?

Consider the following sequence:

1. Update in cluster 1 registry enters `stateProvisionallyAccepted`
2. Failover to cluster 2, leaving `state = stateProvisionallyAccepted`
3. Update is accepted in cluster 2
4. Failback to cluster 1
5. **Bug:** Now the update is in `stateProvisionallyAccepted` but it should be in `stateAccepted` 

Whether or not this is possible depends on whether the rollback function is called on failover, or whether the registry is cleared on failover. But I believe the answer is "no" in both cases currently. So I think this is a possible bug, right?